ansible_runner/interface.py

```diff
@@ -91,7 +93,15 @@ def init_runner(**kwargs):
     event_callback_handler = kwargs.pop('event_handler', None)
     status_callback_handler = kwargs.pop('status_handler', None)
     artifacts_handler = kwargs.pop('artifacts_handler', None)
-    cancel_callback = kwargs.pop('cancel_callback', None)
+    if kwargs.get('cancel_callback'):
+        cancel_callback = kwargs.pop('cancel_callback')
+    # Only the main thread is allowed to set a new signal handler
+    elif threading.current_thread() is threading.main_thread():
+        signal_handler = SignalHandler()
+        cancel_callback = signal_handler.called
```

Comment:
An example of using `threading.Event` to accomplish the same thing.
```suggestion
        signal_event = threading.Event()
        signal.signal(signal.SIGTERM, _signal_shutdown(signal_event))
        signal.signal(signal.SIGINT, _signal_shutdown(signal_event))
        cancel_callback = signal_event.is_set
```

```python
def _signal_shutdown(signal_event):
    signal_event.set()
    
    def inner(number, frame):
        pass
        
    return inner
```

Reply:
Reimplemented signal handler using Threading library.

Reply:
✅

Reply:
needed to convert this to a closure